### PR TITLE
[UPDATE] Updated packages to latest build

### PR DESCRIPTION
**Pull request by NullDivision**
- version incremented to 2.2.5
- babel 5 replaced with 6
- babel presets installed
- babel-eslint updated
- added direct babel-registry
- fbjs updated to version 6
- babel configs moved to package.json

### CODE REVIEW

**Comment by NullDivision** [comment]
Hi. I noticed that the latest version of react-toastr uses an outdated version of fbjs so i went ahead and updated the package.

Everything still works and tests are passing.

File: package.json [hunk 1]
```diff
@@ -69,12 +73,19 @@
...
-    "fbjs": "^0.3.1",
+    "fbjs": "^0.6.0",
     "react": "^0.14.0"
   },
   "dependencies": {
     "classnames": "^2.1.5",
     "react-addons-update": "^0.14.0",
     "react-dom": "^0.14.0"
+  },
+  "babel": {
```
Comment (tomchentw) [inline package.json]: Please put this back to `.babelrc`.

File: package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "__template__gist__": "https://gist.github.com/tomchentw/368a93bb748ad9d576f1#file-package-json",
   "name": "react-toastr",
-  "version": "2.2.4",
+  "version": "2.2.5",
```
Comment (tomchentw) [inline package.json]: Please don't bump commit. Will do that in a release script.

**Comment by tomchentw** [comment]
@NullDivision WOW, thanks for the hard work! It looks great. I've left two code review comments. Please help to fix them and fixup the original commit with the following commit message:

```
feat(package.json): bump fbjs to ^0.6.0

* upgrade to babel@^6
```

File: package.json (hunk 1 above)
Comment (NullDivision) [inline package.json]: I've had issues with .babelrc when importing into projects that use Babel v5. But I put it back into `.babelrc`.

File: package.json (hunk 2 above)
Comment (NullDivision) [inline package.json]: Reverted as requested. :)

**Comment by NullDivision** [comment]
I made the requested changes and left a comment on consideration towards `.babelrc`.

**Comment by tomchentw** [comment]
Thanks! Released [v2.3.0](https://github.com/tomchentw/react-toastr/releases/tag/v2.3.0)

**Comment by tomchentw** [comment]
Just found that @NullDivision had some experience with Facebook's flow. Just wondering that is there any way to run `flow` as a loader step when using `webpack`? It seems that I cannot find a proper way/existing loader plugin for that. But it should be doable?
screen shot 2016-01-05 at 9 46 14 am

**Comment by NullDivision** [comment]
`Flowcheck-loader` does what you want. Provided that you add the `@flow` comment tag and do `flow init` to generate the `.flowconfig`.

**Comment by tomchentw** [comment]
@NullDivision I've checked that, but it seems `flowcheck` is outdated for a while. Maybe I should give [babel-plugin-typecheck](https://github.com/codemix/babel-plugin-typecheck) a try.

**Comment by NullDivision** [comment]
As far as I know that plugin only helps in transpiling type annotations from Flow. I'll have to integrate it into my project as well soon enough so if it's outdated I'll probably end up updating that too. Force of habit :D

If you want to run it at build time you can just add Flow as a devDependency and call it with a conditional operator followed by the build command.

If you want to use it with the webpack-dev-server it'll have to do with the loader.

On the other hand I run it currently on Sublime Text with SublimeLinter and it works great.

**Comment by tomchentw** [comment]
Haha, I just tried and integrated `babel-plugin-typecheck` and `babel-plugin-transform-flow-comments` to my babel6 projects. It works as expected (immediately caught one error) so I'll just them.

**Comment by NullDivision** [comment]
Cool. :) But take into consideration that it adds sizable overhead when transpiling as it adds exceptions to every function call. ...which isn't a bad idea but I'd rather implement my own catches for those :)

**Comment by tomchentw** [comment]
Yup, so I only put it under specific `env`.

```json
{
  "env": {
    "test": {
      "plugins": [
        "transform-runtime",
        "typecheck"
      ],
    },
  },
  "plugins": [
    "transform-flow-comments"
  ],
  "presets": [
    "es2015",
    "stage-0",
    "react"
  ]
}
```

**Pull request merged**